text/0009-golang.md

```diff
@@ -0,0 +1,68 @@
+
+According to [GitHub statistics](https://madnight.github.io/githut/#/pull_requests/2023/3), Go is the third fastest growing language (behind Python and Java) as measured by the number of new Pull Requests.
+
+## Risks and Drawbacks
+
+Introducing Go as one of the supported backend technologies involves the same general concerns as introducing any new technology into FOLIO. The biggest potential risk is the increased development and maintenance complexity as some new FOLIO backend modules will be created with Go which is not immediately well known to existing FOLIO backend developers that have primarily Java background. As explained in the previous section, however, this risk is highly minimized because of the design of Go which has many points of similarity to Java.
+
+Additionally, as history of FOLIO development shows, there exists high level of affinity between specific teams of developers and particular modules and domains. This affinity often results in siloing and can create barriers preventing wider collaboration between various development teams. This siloing exists even within Java-only usage in FOLIO (vert.x vs Spring) because of the high reliance on web programming frameworks. Development in Go has less or no reliance on frameworks -- for instance it's entirely possible to develop FOLIO modules with only Go standard library, Postgres client library and OpenAPI parser library. This more lightweight approach to development can ease some of the cross-module development barriers in FOLIO. 
+
+## Rationale and Alternatives
+
+As explained in the design section, Go is a programming language dedicated to writing backend web services and as such is a perfect fit for FOLIO's microservices-inspired architecture. It is also a language that is increasingly popular with community-driven open source projects thanks to it simplicity and familiar syntax but without the performance tradeoffs that are typical to dynamic or scripting languages.
+
+Go's excellent AOT compilation support puts it ahead of Java for building systems that comprise of many small, containerized workloads.
+
+As such there's no direct alternative to Go available in the FOLIO ecosystem.
+
+## Timing
+
+We propose that Go is added to the list of supported technologies starting from Quesnelia in the form of a pilot project. The pilot will serve as the technology preview with the aim of validating Go's fit for FOLIO backend development. No existing core FOLIO modules will be re-developed using Go during the pilot phase and any new modules written in Go would be considered proof-of-concept modules. 
+
+In the case Go is deemed unsuitable for FOLIO development, any modules developed in Go will be removed from the future FOLIO versions and Go will be dropped as a supported technology.
+
+In the case Go is deemed suitable for FOLIO backend development, it will be accepted to the official technology list without any restrictions starting from Ramsons.
```

Comment:
Since this discusses a go/no-go decision for officially supporting the language (😄), it may be worthwhile to propose some timing for the decision